internal/pkg/api/node/list.go

```diff
@@ -87,6 +90,47 @@ func NodeList(nodeGet *wwapiv1.GetNodeList) (nodeList wwapiv1.NodeList, err erro
 					fmt.Sprintf("%s:=:%s:=:%s:=:%s", n.Id.Print(), f.Field, f.Source, f.Value))
 			}
 		}
+	} else if nodeGet.Type == wwapiv1.GetNodeList_YAML || nodeGet.Type == wwapiv1.GetNodeList_JSON {
+		nodeResp := node.NodeYaml{
+			NodeProfiles: make(map[string]*node.NodeConf),
+			Nodes:        make(map[string]*node.NodeConf),
+		}
+
+		filterNodes := node.FilterByName(nodes, nodeGet.Nodes)
+		for _, nodeInfo := range filterNodes {
+			myNode := node.NewConf()
+			myNode.GetFrom(nodeInfo)
+			if myNode.Kernel == nil || (myNode.Kernel.Args == "" && myNode.Kernel.Override == "" && myNode.Kernel.Version == "") {
```

Comment:
You can replace the checks for Ipmi and Kernel with 
`myNode.Flatten()`
which traverses through the `NodeConf` struct and removes the empty elements